Framework/API/src/LogManager.cpp

```diff
@@ -268,16 +346,23 @@ double LogManager::getPropertyAsSingleValue(
   const auto key = std::make_pair(name, statistic);
   if (!m_singleValueCache.getCache(key, singleValue)) {
     const Property *log = getProperty(name);
-    if (auto singleDouble =
-            dynamic_cast<const PropertyWithValue<double> *>(log)) {
-      singleValue = (*singleDouble)();
-    } else if (auto seriesDouble =
-                   dynamic_cast<const TimeSeriesProperty<double> *>(log)) {
-      singleValue = Mantid::Kernel::filterByStatistic(seriesDouble, statistic);
+    if (convertSingleValue(log, singleValue) ||
+        convertTimeSeriesToDouble(log, singleValue, statistic)) {
+      return singleValue;
+    } else if (const auto stringLog =
+                   dynamic_cast<const PropertyWithValue<std::string> *>(log)) {
+      // Try to lexically cast string to a double
+      try {
+        return std::stod(stringLog->value());
+      } catch (const std::invalid_argument &) {
+        throw std::invalid_argument(
+            "Run::getPropertyAsSingleValue - Property\"" + name +
+            "\"cannot be converted to a numeric value.");
+      }
```

Comment:
I think you are now missing a test for this second exception branch, should that be checked too? You may also want to check the message given when throwing for the string, I think it is missing some spaces :smile: .

Otherwise all looks good.